canonical-web-and-design/maas-ui
# feat(ui): add error handling to machine storage forms

## Done

- Added `useMachineDetailsForm` hook which returns the saving, saved and error state for a single machine performing a single action. It's slightly different to `useMachineActionForm` because it doesn't handle multiple machines processing.
- Added error messages to storage forms

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a ready or allocated machine
- Mount a special filesystem at "/" if there isn't one already
- Try to mount another special filesystem at "/" and check that an error displays and the form does not close
- Do the same for adding/editing partitions

## Fixes

Fixes #1968
